src/Defer/DeferrableDirective.php

```diff
@@ -39,7 +39,7 @@ public function __construct(ExtensionRegistry $extensions)
      */
     public function name(): string
     {
-        return 'deferrable';
+        return self::NAME;
```

Comment:
If you are adding a constant for name, then let's move this to `BaseDirective`, so you won't have to define both the constant and the method. 

Users can still override the method in the rare cases where they need more advanced logic.

Reply:
Unfortunately we cannot guarantee child directives to define a constant.

I think we could change the function to `public static name(): string`?

Reply:
We could define a default for it or just let it fail if they did not define it. This is something we set in `BaseDirective`, so it is just a helper for naming.
The contract still requires an implementation of `name()` which returns a `string`.

Reply:
I think requiring this constant to be set is more magic then just enforcing it via the interface.